pdf/model/xobject.go

```diff
@@ -281,11 +292,74 @@ func NewXObjectImageFromImage(img *Image, cs PdfColorspace, encoder StreamEncode
 		smask.Height = &img.Height
 		smask.ColorSpace = NewPdfColorspaceDeviceGray()
 		xobj.SMask = smask.ToPdfObject()
+	} else {
+		xobj.SMask = xobjIn.SMask
+		xobj.ImageMask = xobjIn.ImageMask
+		if xobj.ColorSpace.GetNumComponents() == 1 {
+			smaskMatteToGray(xobj)
+		}
 	}
 
 	return xobj, nil
 }
 
+// smaskMatteToGray converts to gray the Matte value in the SMask image referenced by `xobj` (if
+// there is one)
+func smaskMatteToGray(xobj *XObjectImage) error {
+	if xobj.SMask == nil {
+		return nil
+	}
+	stream := xobj.SMask.(*PdfObjectStream)
```

Comment:
Might be worth type checking, e.g.
stream, ok := xobj.SMask.(*PdfObjectStream)
if !ok {
  return ErrTypeCheck
}